shell/platform/embedder/embedder.h

```diff
@@ -401,6 +431,11 @@ typedef struct {
   size_t struct_size;
   /// Id of the fbo backing the surface that was presented.
   uint32_t fbo_id;
+  /// NEW: Adding new fields so that present with partial repaint works.
+  /// Rectangle representing the area that the compositor needs to render.
+  FlutterDamage frame_damage;
+  /// Rectangle representing the area that changed since FBO was last used.
+  FlutterDamage buffer_damage;
```

Comment:
Wouldn't we need to know if we're double/triple buffering to compute this on behalf of users? Swapchain length for GL is not always knowable and is highly dependent on the underlying window system, so we may want to just remove this field and leave computing accumulated buffer damage as an optional exercise for embedder authors -- otherwise we'd probably have to add an optional field for users to tell us how many buffers they're going to use in the render config (if this information happens to be accessible to them).

Reply:
That's true! I updated the design document with more details about this discussion as well as comparison between an approach that has the user take care of the accumulated damage vs one that has the rendering backend do it. Considering scalability, the current version of the prototype goes with the approach you suggested.

Reply:
Quick update: this is actually referring to the buffer damage as calculated by joining the frame damage computed by the rendering backend and the existing buffer damage. It is different from the existing damage that is still provided by the user and it needs to be passed to the present callback to allow users to set the damage region before swapping buffers.